gprofiler/profilers/java.py

```diff
@@ -197,6 +204,50 @@ def __str__(self) -> str:
         )
 
 
+class JattachJcmdRunner:
+    def __init__(self, stop_event: Event, jattach_timeout: int):
+        self.stop_event = stop_event
+        self.jattach_timeout = jattach_timeout
+
+    def run(self, process: Process, cmd: str) -> str:
+        return run_process(
+            [jattach_path(), str(process.pid), "jcmd", cmd], stop_event=self.stop_event, timeout=self.jattach_timeout
+        ).stdout.decode()
+
+
+@dataclasses.dataclass
+class JvmFlag:
+    name: str
+    type: str
+    value: str
+    origin: str
+    kind: List[str]
+
+    @classmethod
+    def from_match(cls, match: Match) -> "JvmFlag":
+        # get the flag origin if jvm 9+, otherwise get is the flag from non default origin as described above
+        flag_is_non_default_origin_only_jdk_8 = match.group("flag_is_non_default_origin_only_jdk_8") == ":"
+        flag_origin_jdk_9 = match.group("flag_origin_jdk_9")
+
+        # split the list of space separated flag_kinds as described above
+        flag_kind = match.group("flag_kind").split()
+
+        if flag_is_non_default_origin_only_jdk_8:
+            flag_origin = "non-default"
+        elif flag_origin_jdk_9:
+            flag_origin = flag_origin_jdk_9
+        else:
+            flag_origin = "default"
+
+        return cls(
+            name=match.group("flag_name"),
+            type=match.group("flag_type"),
+            value=match.group("flag_value"),
+            origin=flag_origin,
+            kind=flag_kind,
```

Comment:
```suggestion
            kind=sorted(flag_kind),
```
Prevent accidental cardinality increase.